### PR TITLE
Move scope of coreos namespace to all clusters

### PR DESCRIPTION
**Pull request by darkmuggle**
To enable building of CoreOS images on any cluster via Gangplank we need
to move the cluster scope from boot02 to all the clusters.

The ci-operator has also been granted admin scope permissions to the
coreos namespace. This will allow us to use the ci-operator to create
pods in the cores namespaces via Prow jobs.

Signed-off-by: Ben Howard <ben.howard@redhat.com>

### CODE REVIEW

**Comment by darkmuggle** [comment]
/assign @alvaroaleman 

@alvaroaleman mind a review?

**Comment by alvaroaleman** [comment]
What exactly is this thing doing?

**Comment by darkmuggle** [comment]
@alvaroaleman Gangplank is a pod runner that executes on any Kubernetes cluster and it knows how to setup the pod correctly for CoreOS Assembler. Gangplank discovers secrets and uses simple YAML for running. When run in a pod, assuming it has an appropriate service account, it will create worker pod(s) that run as soon as dependencies are ready (for example, we do a base build, but then need to build an image for AWS and GCP based on that image). 

Our problem with RHCOS is that:
- each build is huge (~60Gb)
- we need KVM access (different depending on the cluster version)
- we have a dozen pipelines and its hard to manage

With Gangplank, we the same spec to do a build on any environment. In the case of Prow, we want to use it for the development and CI builds. However, for the ART builds that use Jenkins and must be built behind the wall for OS images.

Since we have to straddle FCOS using Jenkins and an OCP 4.x cluster, ART using a privileged OCP 3.11 baremetal cluster,  local development builds via podman and the desires for Prow builds, Gangplank was created to distill the environment setup using simple rules. 

Basically, its hard to build operating system images that boot anywhere. If we doing container images, the standard prow would do.

**Comment by alvaroaleman** [comment]
@darkmuggle build02 is the only cluster that supports kvm.
/cc @cgwalters

**Comment by cgwalters** [comment]
> each build is huge (~60Gb)

Only tangentially related but IMO we don't need to build all artifacts on every PR though, I think it'd be better instead to have a flow where we default to qemu but have /test e2e-aws and /test e2e-gcp that opt-in to that.  There's actually also the huge bifurcation here around testing bootimages versus containers.  In most cases I think what we want is e.g. `/test e2e-aws-upgrade` that just makes a new oscontainer+release image and passes that as an upgrade target, i.e. no new AMI.

Also on this topic a `test kola-metal` would make sense that runs through the whole `metal*` bits and `kola testiso`.

And then for big updates we can do `/test all` and just try everything.

**Comment by cgwalters** [comment]
xref https://issues.redhat.com/browse/DPTP-2115

**Comment by alvaroaleman** [comment]
Does this now need hardware virtualization, yes or no?

**Comment by cgwalters** [comment]
> Does this now need hardware virtualization, yes or no?

It's a mix.  The core build process does - for example when we generate artifacts like the "oscontainer", qcow2, ISO, GCP/Azure/AWS images.  And we want to continue to use qemu to do basic sanity checks on generated images.  But, if we e.g. upload a test AMI from that build, then we can schedule further pods that run tests on it and *those* pods don't need KVM, any more than the regular cluster-launching e2e test pods do when they launch a remote cluster.

**Comment by cgwalters** [comment]
BTW one thing I just connected/understood is that the `oc adm release new` bit is currently a [special step](https://github.com/openshift/ci-tools/blob/master/pkg/steps/release/create_release.go) - and our desired flow here in some cases is actually to create a custom `machine-os-content` and then use that step too.  So one possibility here seems to be extending the CI tools steps for that. 

But that path conflicts a bit with our desire to easily run these steps *outside* of Prow so we can test things outside of submitting pull requests - a goal of Gangplank is that it also works with e.g. local podman.

**Review by alvaroaleman — APPROVED** [review]
